usb_mass_storage.go

```diff
@@ -269,16 +270,48 @@ func rpcMountWithStorage(filename string, mode VirtualMediaMode) error {
 		return fmt.Errorf("failed to get file info: %w", err)
 	}
 
-	err = setMassStorageImage(fullPath)
+	// Handle XZ compressed images
+	if strings.HasSuffix(filename, ".img.xz") {
+		logger.Info("Mounting compressed XZ image")
+
+		// Create temporary file for decompressed image
+		decompressedPath := filepath.Join(imagesFolder, strings.TrimSuffix(filename, ".xz")+".temp")
+		defer os.Remove(decompressedPath) // Clean up temp file after mounting
+
+		err = decompressXZImage(fullPath, decompressedPath)
+		if err != nil {
+			return fmt.Errorf("failed to decompress XZ image: %w", err)
+		}
+
+		// Mount the decompressed image
+		err = mountImage(decompressedPath)
+		if err != nil {
+			return fmt.Errorf("failed to mount decompressed image: %w", err)
+		}
+
+		currentVirtualMediaState = &VirtualMediaState{
+			Source:   Storage,
+			Mode:     mode,
+			Filename: filename,
+			Size:     fileInfo.Size(),
+		}
+
+		return nil
```

Comment:
This entire block is a copy/paste of the next code bit, so can't we just fall-though to that after adjusting the path
```go
fullPath = decompressedPath
```